lib/sda/overlay.go

```diff
@@ -28,17 +28,26 @@ type Overlay struct {
 	entityListLock sync.Mutex
 	// cache for relating token(~Node) to TreeNode
 	cache TreeNodeCache
+
+	// TreeNodeInstance part
+	instances     map[TokenID]*TreeNodeInstance
+	instancesInfo map[TokenID]bool
+	instancesLock sync.Mutex
+	pis           map[TokenID]ProtocolInstance
```

Comment:
if all other names are written out, I propose to use `protocolInstances` here, too. Even though it's longer...